service/co/hotwax/bi/DimensionServices.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<services xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://moqui.org/xsd/service-definition-3.xsd">
+    <!--    <service verb="load" type="script" noun="DateDimension" location="component://bi/src/loadDateDimension.groovy">-->
+    <service verb="load" noun="DateDayDimension">
+        <in-parameters>
+            <parameter name="fromDate" type="Timestamp" format="yyyy-MM-dd" required="true">
+                <description>From date to generate the date dimension</description>
+            </parameter>
+            <parameter name="thruDate" type="Timestamp" format="yyyy-MM-dd" required="true">
+                <description>Thru date to generate the date dimension</description>
+            </parameter>
+        </in-parameters>
+        <actions>
+            <set field="dateDimension" type="String" value="moqui.olap.DateDayDimension" />
+            <script>
+                import java.text.SimpleDateFormat
+
+                SimpleDateFormat monthNameFormat = new SimpleDateFormat("MMMM")
+                SimpleDateFormat dayNameFormat = new SimpleDateFormat("EEEE")
+                SimpleDateFormat dayDescriptionFormat = new SimpleDateFormat("MMMM d, yyyy")
+                SimpleDateFormat yearMonthDayFormat = new SimpleDateFormat("yyyy-MM-dd")
+                SimpleDateFormat yearMonthFormat = new SimpleDateFormat("yyyy-MM")
+
+                calendar = Calendar.getInstance()
+                calendar.setTime(fromDate)
+                calendar.set(Calendar.HOUR, 0)
+                calendar.set(Calendar.MINUTE, 0)
+                calendar.set(Calendar.SECOND, 0)
+                calendar.set(Calendar.MILLISECOND, 0)
+            </script>
+            <set field="currentDate" type="Date" from="new java.sql.Date(calendar.getTimeInMillis())"/>
+            <while condition="currentDate.compareTo(thruDate) &lt;= 0">
+                <entity-find-one entity-name="${dateDimension}" value-field="dateValue">
```

Comment:
entity-find-one should be done using PKs only.
If we are not using PKs, then do entity find and get first.

Reply:
I have checked `entity-find-one` and will ignore checks if the PK is there in the checklist otherwise It will always fetch the first record based on the checks.